algorithms/tree_algorithm.py

```diff
@@ -0,0 +1,187 @@
+from supporting_components.graph import Graph, Vertex
+from math import factorial
+
+
+def is_tree(G: "Graph"):
+    """
+    Determines if graph G is a tree based on the Theorem that the amount of edges is one less than the amount of vertices
+    :return: Boolean if graph G is a tree or not
+    """
+    return len(G.edges) == len(G.vertices) - 1
+
+
+def trees_count_isomorphisms(T1: "Graph", T2: "Graph", count_flag: "Bool"):
+    """
+    :param T1, T2: Graphs that are trees (so is_tree(T1) and is_tree(T2) return True)
+    :param count_flag: Flag that determines what return argument is
+    :return: If count_flag is False, the return argument is a boolean indicating if T1 and T2 are isomorphic or not
+    If count_flag is True, the return argument is the amount of isomorphisms between T1 and T2
+    """
+    # Determine the root of both trees
+    root_T1 = __root_label(T1)
+    root_T2 = __root_label(T2)
+
+    # Tree can have one or two roots, if amount of roots is not equal, not isomorphic
+    if len(root_T1) != len(root_T2):
+        if count_flag:
+            return 0
+        return False
+
+    # If there are two roots, the amount of isomorphisms is the addition of isomorphisms from the two mappings
+    # of roots to each other. Because the maximum amount of roots is 2, all roots of T1 are mapped to one root of T2.
+    # Root levels are overwritten in the second cycle.
+    total_isomorphisms = 0
+    # for i_root in range(len(root_T1)):
+    for i_root in [0]:
```

Comment:
dit is een for loop met 1 iteratie, kan worden vervangen door i_root = 0

Reply:
Die code die gecommend was moest hiervoor in de plaats komen, dit was voor mezelf een testje wat ik nog niet terugveranderd had